RFS/src/main/java/com/rfs/RfsCreateSnapshot.java

```diff
@@ -0,0 +1,92 @@
+package com.rfs;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+import com.rfs.cms.CmsClient;
+import com.rfs.cms.OpenSearchCmsClient;
+import com.rfs.common.ConnectionDetails;
+import com.rfs.common.Logging;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.SnapshotCreator;
+import com.rfs.common.TryHandlePhaseFailure;
+import com.rfs.common.S3SnapshotCreator;
+import com.rfs.worker.GlobalState;
+import com.rfs.worker.SnapshotRunner;
+
+public class RfsCreateSnapshot {
+    private static final Logger logger = LogManager.getLogger(RfsCreateSnapshot.class);
+
+    public static class Args {
+        @Parameter(names = {"--snapshot-name"}, description = "The name of the snapshot to migrate", required = true)
+        public String snapshotName;
+
+        @Parameter(names = {"--s3-repo-uri"}, description = "The S3 URI of the snapshot repo, like: s3://my-bucket/dir1/dir2", required = true)
+        public String s3RepoUri;
+
+        @Parameter(names = {"--s3-region"}, description = "The AWS Region the S3 bucket is in, like: us-east-2", required = true)
+        public String s3Region;
+
+        @Parameter(names = {"--source-host"}, description = "The source host and port (e.g. http://localhost:9200)", required = true)
+        public String sourceHost;
+
+        @Parameter(names = {"--source-username"}, description = "Optional.  The source username; if not provided, will assume no auth on source", required = false)
+        public String sourceUser = null;
+
+        @Parameter(names = {"--source-password"}, description = "Optional.  The source password; if not provided, will assume no auth on source", required = false)
+        public String sourcePass = null;
+
+        @Parameter(names = {"--target-host"}, description = "The target host and port (e.g. http://localhost:9200)", required = true)
+        public String targetHost;
+
+        @Parameter(names = {"--target-username"}, description = "Optional.  The target username; if not provided, will assume no auth on target", required = false)
+        public String targetUser = null;
+
+        @Parameter(names = {"--target-password"}, description = "Optional.  The target password; if not provided, will assume no auth on target", required = false)
+        public String targetPass = null;
+
+        @Parameter(names = {"--log-level"}, description = "What log level you want.  Default: 'info'", required = false, converter = Logging.ArgsConverter.class)
```

Comment:
Does the conversion work from string to enum here?  This is way more powerful with these phases as separate commands now.  Longer term, we need to work out a unified strategy for all of our log level management, but this is a great first start.